taichi/backends/vulkan/vulkan_api.h

```diff
@@ -199,24 +212,19 @@ class VulkanCommandBuilder {
 
   VkCommandBuffer build();
 
+  void dispatch(const VulkanPipeline &pipeline, int group_count_x);
+
+  void copy(VkBuffer src_buffer,
+            VkBuffer dst_buffer,
+            VkDeviceSize size,
+            VulkanCopyBufferDirection direction);
+
  protected:
   // VkCommandBuffers are destroyed when the underlying command pool is
   // destroyed.
   // https://vulkan-tutorial.com/Drawing_a_triangle/Drawing/Command_buffers#page_Command-buffer-allocation
   VkCommandBuffer command_buffer_{VK_NULL_HANDLE};
-};
-
-class VulkanComputeCommandBuilder : public VulkanCommandBuilder {
- public:
-  using VulkanCommandBuilder::VulkanCommandBuilder;
-
-  void append(const VulkanPipeline &pipeline, int group_count_x);
-};
-
-enum class VulkanCopyBufferDirection {
-  H2D,
-  D2H,
-  // D2D does not have a use case yet
+  VkDevice device;  // do not own
```

Comment:
We should probably get a linter..

```suggestion
  VkDevice device_;  // do not own
```